bug 1483670: accept unicode from update endpoints

mozbhearsum opened the pull request:
This is a follow-up to #634 and #744 that allows us to accept unicode in all of the string fields for the /update endpoints. Because we receive so many weird requests, I don't think we can get away with throwing errors for them all. This patch accepts it - but replaces the unicode characters with "?". In a far flung future where we support unicode at the db and other balrog layers, we should be able to simply strip this out, or perhaps encode to utf-8 instead. In addition to the unit tests included, I also grabbed a bunch of URLs from Sentry and verified that I got 200s from them after applying this patch.

Finally, I copied over the UnicodeEncodeError handler from the admin app to public. Although the /update endpoints don't support unicode, a couple of fields on the public API do - so that handler is still useful for them.

nthomas-mozilla commented:
This looks good to me. A closely related issue is tracebacks like at https://sentry.prod.mozaws.net/operations/prod-public/issues/4339330/, where there is a bogus unicode query parameter. Would it be worth addressing this at the same time ?

mozbhearsum commented:
Definitely. I'll update this PR to handle them.

mozbhearsum commented:
OK, I've updated this to handle the query args as best as possible (accept unicode as values, and verify that unicode in field names doesn't throw a 500).

I also updated this to fix some issues found by the new Python 3 tests (thanks @allan-silva!). I think there may be more work to do before we upgrade to Python 3, but this should be OK for now.

nthomas-mozilla commented on version.txt:
```diff
@@ -1 +1 @@
-2.53
\ No newline at end of file
+2.53
```
Ah, you're fixing up my removal of the newline, never mind!
